rsocket/internal/ManageableConnection.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/futures/SharedPromise.h>
+#include <folly/io/async/EventBase.h>
+
+namespace rsocket {
+
+class MockManageableConnection;
+enum class StreamCompletionSignal;
+
+/// Utility class that enables RSocketConnectionManager to handle
+/// lifetime of RSocketStateMachine objects.
+class ManageableConnection {
```

Comment:
Does this need to be its own separate class? It's just a wrapper around a promise at this point.

Reply:
Better to be simple :)

Reply:
Why is introducing a new class and inheriting from it simpler than just putting the promise to indicate "is the connection closed yet" in RSocketStateMachine directly?